BridgeSDK/BridgeAPI/SBBActivityManager.m

```diff
@@ -116,12 +116,18 @@ - (NSArray *)filterTasks:(NSArray *)tasks forDaysAhead:(NSInteger)daysAhead andD
 
 - (NSArray *)filterTasks:(NSArray *)tasks scheduledFrom:(NSDate *)startDate to:(NSDate *)endDate
 {
+    // It's in the date range [startDate, endDate) if:
+    //  - it never expires or expires on or after startDate AND
+    //  - it hasn't been marked finished or was marked finished on or after startDate AND
+    //  - it was scheduled before endDate
```

Comment:
Okay, now I'm confused?  Why are you checking finished on here but not above? Also, you have an assumption here that the date range isn't fully within the past. For instance a UI where today is filtered out (for whatever reason) would fail this test. For example, if the task was finished today but the requested range only includes yesterday.